test/framework/machinedeployment_helpers.go

```diff
@@ -137,3 +141,137 @@ func DiscoveryAndWaitForMachineDeployments(ctx context.Context, input DiscoveryA
 	}
 	return machineDeployments
 }
+
+// UpgradeMachineDeploymentAndWaitInput is the input type for UpgradeMachineDeploymentAndWait.
+type UpgradeMachineDeploymentAndWaitInput struct {
```

Comment:
```suggestion
type UpgradeMachineDeploymentsAndWaitInput struct {
```
because it gets an array of machine deployments in input